lighthouse-core/config/config-helpers.js

```diff
@@ -17,6 +18,21 @@ const validation = require('../fraggle-rock/config/validation.js');
 /** @typedef {typeof import('../audits/audit.js')} Audit */
 /** @typedef {InstanceType<GathererConstructor>} Gatherer */
 
+function isBundledEnvironment() {
+  // If we're in DevTools or LightRider, we are definitely bundled.
+  // TODO: refactor and delete `global.isDevtools`.
+  if (global.isDevtools || global.isLightrider) return true;
+
+  try {
+    // Not foolproof, but `lighthouse-logger` is a dependency of lighthouse that should always be resolvable.
+    // `require.resolve` will only throw in atypical/bundled environments.
+    require.resolve('lighthouse-logger');
```

Comment:
i cri in esm evry time

( I don't have an alternative suggestion )

Reply:
except for: making build-bundle inject a `globalThis.isBundled`...

Reply:
we should be able to use createRequire/import.meta.require/require.resolve combo based on environment, though detection will be annoying :)